### PR TITLE
Extract private apis

### PR DESCRIPTION
**Pull request by ms-ati**
In order to increase the clarity and maintainability of the codebase, extract
internal private apis from the main MemoWise module. This should also make
it easier to refactor private api code.

**Before merging:**

- [x] Copy the latest benchmark results into the `README.md` and update this PR
~[ ] If this change merits an update to `CHANGELOG.md`, add an entry following Keep a Changelog [guidelines](https://keepachangelog.com/en/1.0.0/) with [semantic versioning](https://semver.org/)~

### CODE REVIEW

**Comment by ms-ati** [comment]
Re: #160

File: lib/memo_wise/internal_api.rb [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+    def validate_memo_wised!(method_name)
+      klass = target.instance_of?(Class) ? target.singleton_class : target.class
+      original_memo_wised_name = :"_memo_wise_original_#{method_name}"
+
+      unless klass.private_method_defined?(original_memo_wised_name)
+        raise ArgumentError, "#{method_name} is not a memo_wised method"
+      end
+    end
+
+    # TODO: Parameter validation for presetting values
```
Comment (jemmaissroff) [inline lib/memo_wise/internal_api.rb:236]: I wonder if we should just remove this placeholder for now? As far as I know, we won't do this for V1, so probably okay to just add it back when we need it?

**Review by jemmaissroff — APPROVED** [review]

File: lib/memo_wise/internal_api.rb (hunk 1 above)
Comment (ms-ati) [reply lib/memo_wise/internal_api.rb:236]: Ok, I'll just go ahead and remove this now then

Comment (ms-ati) [reply lib/memo_wise/internal_api.rb:236]: Ok, despite what I wrote in slack, I'll go ahead and just remove this now

**Pull request merged**